src/main/java/com/google/sps/data/BusinessesService.java

```diff
@@ -110,4 +126,102 @@ private void addListingToBusinesses(PlacesSearchResult place) {
     allBusinesses.add(new Listing(name, formattedAddress, 
           placeLocation, rating, photos, types));
   }
+
+  public void checkNumberOfLocationsOfBusiness() {
+    GeoApiContext context = new GeoApiContext.Builder()
+            .apiKey(KEY)
+            .build();
+    try {
+      for(int i = 0; i<allBusinesses.size();i++){
+        currentBusiness = allBusinesses.get(i);
+        TextSearchRequest request = new TextSearchRequest(context)
+                                            .query(currentBusiness.getName())
+                                            .location(latLng)
+                                            .radius(50000);
+        similarBusinessesInTheArea = request.await();
+        if(similarBusinessesInTheArea.results.length > 1){
+          checkBusinessThroughLinkedin(currentBusiness.getName());
+        }
```

Comment:
Could this also be a forEach loop? Since allBusinesses is a linkedList, the allBusinesses.get(i) function will have to iterate to the ith position of the list for each business. If this is a forEach, you would be able to reference the current business directly.

Reply:
I made it a while loop instead, so that I can remove the business when I know that it is a big one.